### PR TITLE
Run parser plugins for top-level unknown elements

### LINKED ISSUE #494: parser plugins should parse top-level unknown and void elements
The Section parser first infers a section type for a given element before attempting to parse its child nodes, and it only runs its parser plugins against the child nodes. This means that HTML with root-level elements do not get run through the plugins. This happens (in slightly different ways) with both void elements and elements with childNodes:
- `parse('<video>...</video>')` — plugins skip `<video>`
- `parse('<video>')` — plugins skip  `video`

### PR DESCRIPTION
**Pull request by kevinansfield**
closes #494
- run parser plugins for the top-level section elements
    - do not continue parsing children elements if a parser marks the top-level element as finished
- update tests to double-check the output for top-level elements is unchanged

@bantic I'd be interested in a quick review and any pointers for any edge cases/tests you think may need adding 😄

### CODE REVIEW

File: src/js/parsers/section.js [hunk 1]
```diff
@@ -69,14 +69,18 @@ class SectionParser {
 
     this._updateStateFromElement(element);
 
-    let childNodes = isTextNode(element) ? [element] : element.childNodes;
+    let finished = this.runPlugins(element);
```
Comment (mixonic) [inline src/js/parsers/section.js]: Given an `element` here that is a text node, wouldn't the add call here mean that text node will be passed to `this.runPlugins` twice? Once here and once in `pardeNode`.

File: tests/unit/editor/editor-test.js [hunk 2]
```diff
@@ -354,7 +354,7 @@ test('editor parses HTML post using parser plugins', (assert) => {
   editor = new Editor({html, parserPlugins: [parserPlugin]});
   assert.ok(!!editor.post, 'editor loads post');
 
-  assert.deepEqual(seenTagNames, ['TEXTAREA', 'IMG']);
+  assert.deepEqual(seenTagNames, ['P', 'TEXTAREA', 'IMG']);
```
Comment (mixonic) [inline tests/unit/editor/editor-test.js]: Is this characterized as a bugfix or as a breaking change?

File: src/js/parsers/section.js (hunk 1 above)
Comment (kevinansfield) [reply src/js/parsers/section.js]: Good catch 😄 Pushed a fix

File: tests/unit/editor/editor-test.js (hunk 2 above)
Comment (kevinansfield) [reply tests/unit/editor/editor-test.js]: I would characterise it as a bugfix because the old behaviour meant elements that were expected to go through parser plugins were never seen. I suppose it may be a breaking change if a consumer has built parser plugins to work around the buggy behaviour, I haven't been able to think up an example of that yet 🤔

Comment (mixonic) [reply tests/unit/editor/editor-test.js]: We talked about this and considered some examples. It seems like it wouldn't break any code we looked at in our quick review, so I'm going to call this a bugfix 👍

**Comment by sdhull** [comment]
Looking forward to trying `parserPlugins` with this update, thanks @kevinansfield!! 

We found that when our users pasted in `<b>` or `<i>` tags, the could not un-bold or un-italicize that text because our toolbar toggles `strong` and `em` tags. I tried to add parser plugins that would convert `b` and `i` tags to `strong` and `em` respectively however I found that my plugins were never called for these tags.

I've worked around this by using computed properties (eg toggle `boldTag` instead of `"strong"`) in Ember but it's not an ideal solution imo.
